server/resp/src/main/java/org/infinispan/server/resp/scripting/LuaTaskEngine.java

```diff
@@ -0,0 +1,380 @@
+         LuaContext luaCtx = pool.borrow();
+         ScriptWithMetadata script;
+         try {
+            script = scriptingManager.getScriptWithMetadata(scriptName(sha.toUpperCase()));
+         } catch (CacheException e) {
+            throw new RuntimeException("NOSCRIPT No matching script. Please use EVAL.");
+         }
+         LuaCode code = LuaCode.fromScript(script);
+         luaCtx.registerScript(code);
+         runScript(luaCtx, handler, ctx, code, keys, args, flags);
+         return luaCtx;
+      }, "evalsha").thenApplyAsync(luaCtx -> {
+         // Process the lua object on the stack and send it to the actual writer
+         luaToResp(luaCtx.lua, handler);
+         // Pop the error handler
+         luaCtx.lua.pop(1);
+         pool.returnToPool(luaCtx);
+         return null;
+      }, ctx.channel().eventLoop());
+   }
+
+   private void runScript(LuaContext luaCtx, Resp3Handler handler, ChannelHandlerContext ctx, LuaCode script, String[] keys, String[] args, long flags) {
+      luaCtx.handler = handler;
+      luaCtx.ctx = ctx;
+      luaCtx.flags = flags;
+      luaCtx.flags |= script.flags();
+      ResponseWriter oldWriter = handler.writer(new LuaResponseWriter(luaCtx.lua));
+      try {
+         TransactionContext.startTransactionContext(ctx);
+         runScript(luaCtx.lua, script, keys, args);
+      } catch (Throwable t) {
+         throw new RuntimeException(t);
+      } finally {
+         TransactionContext.endTransactionContext(ctx);
+         handler.writer(oldWriter);
+      }
+   }
+
+   /**
+    * Convert the response found on the Lua stack to a RESP response
+    */
+   private void luaToResp(Lua lua, Resp3Handler handler) {
+      try {
+         lua.checkStack(4);
+      } catch (RuntimeException e) {
+         handler.writer().customError("reached lua stack limit");
+         lua.pop(1);
+         return;
+      }
+      switch (lua.type(-1)) {
+         case STRING:
+            handler.writer().string(lua.toString(-1));
+            break;
+         case BOOLEAN:
+            handler.writer().booleans(lua.toBoolean(-1));
+            break;
+         case NUMBER:
+            handler.writer().integers((long) lua.toNumber(-1));
+            break;
+         case TABLE:
+            // Is it an error ?
+            lua.push("err");
+            lua.rawGet(-2);
+            if (lua.type(-1) == Lua.LuaType.STRING) {
+               lua.pop(1); // pop the error message
+               ErrorInfo errorInfo = extractErrorInformation(lua);
+               handler.writer().error("-" + errorInfo.message);
+               lua.pop(1); // pop the result table
+               return;
+            }
+            lua.pop(1);
+
+            // Is it a simple status ?
+            lua.push("ok");
+            lua.rawGet(-2);
+            if (lua.type(-1) == Lua.LuaType.STRING) {
+               String ok = lua.toString(-1).replaceAll("[\\r\\n]", " ");
+               handler.writer().string(ok);
```

Comment:
These characters should be fine with the `string` method. It could be problematic for a simple string.